RecoLocalMuon/CSCRecHitD/src/CSCHitFromStripOnly.cc

```diff
@@ -472,11 +469,69 @@ void CSCHitFromStripOnly::findMaxima(const CSCDetId& id) {
           break;
         }
       }
-      theConsecutiveStrips.push_back(numberOfConsecutiveStrips);
-    }
-  }
-}
-//
+
+      bool additional_maxima_found = false;
+      // search for additional maxima if:
+      // - hit is closer than 3 strips from the edge
+      // - enough consecutive strips with signal
+      // - strip charge distribution looks abnormal
+
+      if (i > 2 && i < (thePulseHeightMap.size() - 3) && numberOfConsecutiveStrips > 3) {
```

Comment:
thePulseHeightMap.size() is an unsigned, as well as i . It is then safer avoiding differences with unsigned:
```suggestion
      if (i > 2 && i  +3< thePulseHeightMap.size() && numberOfConsecutiveStrips > 3) {
```